jni/src/faiss_wrapper.cpp

```diff
@@ -509,21 +510,17 @@ void buildFilterIdsBitMap(const int* filterIds, int filterIdsLength, uint8_t* bi
     }
 }
 
-os_faiss::MultiVectorResultCollectorFactory* buildResultCollectorFactory(knn_jni::JNIUtilInterface * jniUtil, JNIEnv *env, jintArray parentIdsJ) {
-    if (parentIdsJ == nullptr) {
-        return nullptr;
-    }
+std::unique_ptr<faiss::IDGrouperBitmap> buildIDGrouperBitmap(knn_jni::JNIUtilInterface * jniUtil, JNIEnv *env, jintArray parentIdsJ, std::vector<uint64_t>* bitmap) {
     int *parentIdsArray = jniUtil->GetIntArrayElements(env, parentIdsJ, nullptr);
     int parentIdsLength = jniUtil->GetJavaIntArrayLength(env, parentIdsJ);
-    auto* parent_id_filter = new FixedBitSet(parentIdsArray, parentIdsLength);
```

Comment:
we should pass parentIdsArray and parentIdsLength directly to this function rather than pass JNIUtilInterface, JNIEnv, parentIdsJ

Reply:
`faiss_wrapper` already has methods which uses JNI objects. Therefore, I moved the core logic into a new file `faiss_util` which will contains all code agnostic to JNI.